Purchases/Attribution/RCAttributionFetcher.m

```diff
@@ -164,22 +136,54 @@ - (void)postAttributionData:(NSDictionary *)data
     }
 }
 
-- (void)postAppleSearchAdsAttributionCollection {
+- (BOOL)isAuthorizedToPostSearchAds {
+#if APP_TRACKING_TRANSPARENCY_REQUIRED
+    if (@available(iOS 14, macos 11, tvos 14, *)) {
+        NSOperatingSystemVersion minimumOSVersionRequiringAuthorization = { .majorVersion = 14, .minorVersion = 5, .patchVersion = 0 };
+
+        BOOL needsTrackingAuthorization = [self.systemInfo isOperatingSystemAtLeastVersion:minimumOSVersionRequiringAuthorization];
+
+        Class<FakeATTrackingManager> _Nullable trackingManagerClass = [self.attributionFactory trackingManagerClass];
+        if (!trackingManagerClass && needsTrackingAuthorization) {
+            RCWarnLog(RCStrings.attribution.search_ads_attribution_cancelled_missing_att_framework);
+            return NO;
+        }
+        NSInteger authorizationStatus = [trackingManagerClass trackingAuthorizationStatus];
+        BOOL authorized = (needsTrackingAuthorization && authorizationStatus == FakeATTrackingManagerAuthorizationStatusAuthorized)
+                          || (!needsTrackingAuthorization && (authorizationStatus == FakeATTrackingManagerAuthorizationStatusAuthorized
+                                                              || authorizationStatus == FakeATTrackingManagerAuthorizationStatusNotDetermined));
+        if (!authorized) {
+            RCWarnLog(RCStrings.attribution.search_ads_attribution_cancelled_not_authorized);
+            return NO;
+        }
+
+    }
+#endif
+    return YES;
+}
+
+- (void)postAppleSearchAdsAttributionIfNeeded {
+    if (!self.isAuthorizedToPostSearchAds) {
+        return;
+    }
+
     NSString *latestNetworkIdAndAdvertisingIdSentToAppleSearchAds = [self
         latestNetworkIdAndAdvertisingIdentifierSentForNetwork:RCAttributionNetworkAppleSearchAds];
-    if (latestNetworkIdAndAdvertisingIdSentToAppleSearchAds == nil) {
-        [self adClientAttributionDetailsWithCompletionBlock:^(NSDictionary<NSString *, NSObject *> *_Nullable attributionDetails,
-                                                              NSError *_Nullable error) {
-            NSArray *values = [attributionDetails allValues];
-
-            bool hasIadAttribution = values.count != 0 && [values[0][@"iad-attribution"] boolValue];
-            if (hasIadAttribution) {
-                [self postAttributionData:attributionDetails
-                              fromNetwork:RCAttributionNetworkAppleSearchAds
-                         forNetworkUserId:nil];
-            }
-        }];
+    if (latestNetworkIdAndAdvertisingIdSentToAppleSearchAds != nil) {
+        return;
```

Comment:
this stuff is pretty much the same, I just updated to use early returns because it seemed clearer